hbase/src/main/java/org/apache/zeppelin/hbase/HbaseInterpreter.java

```
@@ -0,0 +1,148 @@
+import java.util.Properties;
+
+/**
+ * Support for Hbase Shell. All the commands documented here
+ * https://wiki.apache.org/hadoop/Hbase/Shell is supported.
+ *
+ * Requirements:
+ * Hbase Shell should be installed on the same machine. To be more specific, the following dir.
+ * should be available: https://github.com/apache/hbase/tree/master/hbase-shell/src/main/ruby
+ * Hbase Shell should be able to connect to the Hbase cluster from terminal. This makes sure
+ * that the client is configured properly.
+ *
+ * The interpreter takes 3 config parameters:
+ * hbase.home: Root dir. where hbase is installed. Default is /usr/lib/hbase/
+ * hbase.ruby.sources: Dir where shell ruby code is installed.
+ *                          Path is relative to hbase.home. Default: lib/ruby
+ * hbase.irb.load: (Testing only) Default is true.
+ *                      Whether to load irb in the interpreter.
+ */
+public class HbaseInterpreter extends Interpreter {
+  private Logger logger = LoggerFactory.getLogger(HbaseInterpreter.class);
+  private ScriptingContainer scriptingContainer;
+
+  private StringWriter writer;
+
+  static {
+    Interpreter.register("hbase", "hbase", HbaseInterpreter.class.getName(),
+        new InterpreterPropertyBuilder()
+            .add("hbase.home", "/usr/lib/hbase/", "Installation dir. of Hbase")
+            .add("hbase.ruby.sources", "lib/ruby",
+                "Path to Ruby scripts relative to 'hbase.home'")
+            .add("hbase.irb.load", "true", "Load hirb. Optional for testing only")
+            .build());
+  }
+
+  public HbaseInterpreter(Properties property) {
+    super(property);
+  }
+
+  @Override
+  public void open() {
+    String hbase_home = getProperty("hbase.home");
+    String ruby_src = getProperty("hbase.ruby.sources");
+    String abs_ruby_src = hbase_home + ruby_src;
+
+    logger.info("Home:" + hbase_home);
+    logger.info("Ruby Src:" + ruby_src);
+
+    Properties props = System.getProperties();
```

Sorry. Completely missing the point here. I check if there is no hbase directory and I am throwing an InterpreterException. I am doing more than just logging the value. Can you point to examples ? Isnt it better to throw the exception in open method if the parameter is wrong ?